.github/scripts/cleanup-cluster.sh

```diff
@@ -52,5 +52,14 @@ for ns in $namespaces; do
         break
     fi
     echo "deleting namespaces: $ns"
+
+    # Remove finalizers from the namespace if present
+    kubectl get namespace $ns -o json | jq '.spec.finalizers = []' >temp-namespace.json
+    kubectl replace --raw "/api/v1/namespaces/$ns/finalize" -f ./temp-namespace.json
+
+    # Attempt to delete the namespace
     kubectl delete namespace $ns --ignore-not-found=true
 done
+
+# Cleanup temporary file
+rm -f temp-namespace.json
```

Comment:
if an error occurs above/script exists, is it possible that file is not cleaned up ?